app/src/main/java/com/cmput301f20t21/bookfriends/entities/Request.java

```diff
@@ -1,18 +1,24 @@
 package com.cmput301f20t21.bookfriends.entities;
 
 import com.cmput301f20t21.bookfriends.enums.REQUEST_STATUS;
+import com.google.firebase.firestore.DocumentId;
 
 public class Request {
+    @DocumentId
     private String id;
     private String requester;
     private String bookId;
     private REQUEST_STATUS status;
 
-    public Request(String id, String requester, String bookId, String status) {
+    // empty constructor to be use by Document.toObject()
+    public Request() {
+    }
+
+    public Request(String id, String requester, String bookId, REQUEST_STATUS status) {
         this.id = id;
         this.requester = requester;
         this.bookId = bookId;
-        this.status = REQUEST_STATUS.valueOf(status);
+        this.status = status;
```

Comment:
I'm not sure how status will be converted when using `Document.toObject()`

Reply:
it works. screenshot of some logging in the `getAllRequestsByUsername` method
![image](https://user-images.githubusercontent.com/25297856/98876848-5fdeb280-243c-11eb-95f0-2aa4a5896d67.png)